extensions/interactions/DragAndDropSort/directives/DragAndDropSort.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Directive for the DragAndDropSort interaction.
+ */
+
+oppia.constant('DRAG_AND_DROP_UI', 'true');
+
+oppia.directive('oppiaInteractiveDragAndDropSort', [
+  'dragAndDropSortRulesService', 'HtmlEscaperService',
+  'UrlInterpolationService',
+  function(
+      dragAndDropSortRulesService, HtmlEscaperService,
+      UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        onSubmit: '&',
+        // This should be called whenever the answer changes.
+        setAnswerValidity: '&'
+      },
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/interactions/DragAndDropSort/directives/' +
+        'drag_and_drop_sort_interaction_directive.html'),
+      controller: [
+        '$scope', '$attrs', 'UrlService', 'EVENT_PROGRESS_NAV_SUBMITTED',
+        'DRAG_AND_DROP_UI', function(
+            $scope, $attrs, UrlService, EVENT_PROGRESS_NAV_SUBMITTED) {
```

Comment:
You didn't add `DRAG_AND_DROP_UI` here, so it's not getting the value.

Reply:
Yes, I forgot to add it here. After that how can I enable the UI?

Reply:
then you'll want do something like:
`$scope.enableDragDropUI = DRAG_AND_DROP_UI`

Reply:
and reference `enableDragDropUI` in the html file.